src/components/input-number/input-number.tsx

```diff
@@ -0,0 +1,884 @@
+
+  /**
+   * Specifies the minimum length of text for the component's value.
+   *
+   * @mdn [minlength](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#minlength)
+   */
+  @Prop({ reflect: true }) minLength?: number;
+
+  /**
+   * Specifies the name of the component.
+   *
+   * @mdn [name](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#name)
+   */
+  @Prop({ reflect: true }) name: string;
+
+  /** Specifies the placement of the buttons. */
+  @Prop({ reflect: true }) numberButtonType?: InputPlacement = "vertical";
+
+  /**
+   * Specifies placeholder text for the component.
+   *
+   * @mdn [placeholder](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#placeholder)
+   */
+  @Prop() placeholder: string;
+
+  /** Adds text to the start of the component. */
+  @Prop() prefixText?: string;
+
+  /**
+   * When true, the value cannot be modified.
+   *
+   * @mdn [readOnly](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/readonly)
+   */
+  @Prop() readOnly = false;
+
+  /** When true, the component must have a value in order for the form to submit. */
+  @Prop() required = false;
+
+  /** Specifies the size of the component. */
+  @Prop({ mutable: true, reflect: true }) scale: Scale = "m";
+
+  /** Specifies the status of the input field, which determines message and icons. */
+  @Prop({ mutable: true, reflect: true }) status: Status = "idle";
+
+  /**
+   * Specifies the granularity that the component's value must adhere to.
+   *
+   * @mdn [step](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/step)
+   */
+  @Prop({ reflect: true }) step?: number | "any";
+
+  /** Adds text to the end of the component.  */
+  @Prop() suffixText?: string;
+
+  /**
+   * @internal
+   */
+  @Prop({ mutable: true, reflect: true }) editingEnabled = false;
+
+  /** The component's value. */
+  @Prop({ mutable: true }) value = "";
+
+  @Watch("value")
+  valueWatcher(newValue: string, previousValue: string): void {
+    if (!this.userChangedValue) {
+      this.setValue({
+        origin: "direct",
+        previousValue,
+        value:
+          newValue == null || newValue == ""
+            ? ""
+            : isValidNumber(newValue)
+            ? newValue
+            : this.previousValue || ""
+      });
+      this.warnAboutInvalidNumberValue(newValue);
+    }
+    this.userChangedValue = false;
+  }
+
+  @Watch("icon")
+  updateRequestedIcon(): void {
+    this.requestedIcon = setRequestedIcon({}, this.icon, "number");
+  }
+
+  //--------------------------------------------------------------------------
+  //
+  //  Private Properties
+  //
+  //--------------------------------------------------------------------------
+
+  labelEl: HTMLCalciteLabelElement;
+
+  formEl: HTMLFormElement;
+
+  defaultValue: InputNumber["value"];
+
+  inlineEditableEl: HTMLCalciteInlineEditableElement;
+
+  /** number text input element for locale */
+  private childNumberEl?: HTMLInputElement;
```

Comment:
In the spirit of this comment: https://github.com/Esri/calcite-components/pull/4870/files#r916327318 it might make sense to rename this to `childEl` too.  What do you think?